dds/idl/typeobject_generator.cpp

```diff
@@ -526,870 +526,22 @@ operator<<(std::ostream& out, const OpenDDS::XTypes::TypeIdentifier& ti)
   return out;
 }
 
-namespace {
-template<typename T>
-std::string getTypeName();
-
-template<>
-std::string getTypeName<OpenDDS::DCPS::String>()
-{
-  return "String";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AppliedAnnotationParameterSeq>()
-{
-  return "XTypes::AppliedAnnotationParameterSeq";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AnnotationParameterValue>()
-{
-  return "XTypes::AnnotationParameterValue";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AppliedBuiltinMemberAnnotations>()
-{
-  return "XTypes::AppliedBuiltinMemberAnnotations";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AppliedAnnotationSeq>()
-{
-  return "XTypes::AppliedAnnotationSeq";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AppliedVerbatimAnnotation>()
-{
-  return "XTypes::AppliedVerbatimAnnotation";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::AppliedBuiltinTypeAnnotations>()
-{
-  return "XTypes::AppliedBuiltinTypeAnnotations";
-}
-
-template<>
-std::string getTypeName<OpenDDS::XTypes::CompleteTypeDetail>()
+void dump_bytes(const OpenDDS::XTypes::TypeObject& to)
 {
-  return "XTypes::CompleteTypeDetail";
-}
-}
-
-template<typename T>
-std::ostream&
-operator<<(std::ostream& out, const OpenDDS::XTypes::Optional<T>& optional)
-{
-  if (optional.present) {
-    out << "XTypes::Optional<" << getTypeName<T>()  << ">(";
-    if (getTypeName<T>() == "String") {
-      out << "\"" << optional.value << "\"";
-    } else {
-      out << optional.value;
-    }
-    out << ")";
-    return out;
-  } else {
-    return out << "XTypes::Optional<" << getTypeName<T>() << ">()";
+  size_t size = 0;
+  OpenDDS::DCPS::serialized_size(OpenDDS::XTypes::get_typeobject_encoding(), size, to);
+  ACE_Message_Block buffer(size);
```

Comment:
```suggestion
  ACE_Message_Block buffer(OpenDDS::DCPS::serialized_size(OpenDDS::XTypes::get_typeobject_encoding(), to));
```

There's a shortcut for this based on a helper function in Serializer.h.